### PR TITLE
implemented front-end for contacts page

### PR DESCRIPTION
**Pull request by rgao**
#447 

- Created contacts page according to wireframe on Figma
- Includes a small adjustment for responsiveness for mobile-sized screens
- Includes input validation, though needs to be more specific for email

I pushed my backend file for dealing with the github API as well, though it's useless for now. I'm also looking into integrating Sanic's email sending package. 

@adamkendis @jmensch1 
Screenshot from 2020-03-29 02-01-15
Here's what most of the page looks like on my localhost

### CODE REVIEW

**Comment by rgao** [comment]
Alright so I got a ton of linting errors for my jsx files. It seems like I don't have the linting software for front-end installed...

**Comment by jmensch1** [comment]
hey @rgao, you just need to run `npm run lint` and it will tell you how to fix the linting errors

**Comment by rgao** [comment]
fixed the linting errors

File: src/styles/contact/_contactimage.scss [hunk 1]
```diff
@@ -0,0 +1,24 @@
+.contact-cover {
+    height: 55vh;
+    background-image: url('../assets/about311hero860-min.png');
```
[the remark this answers is not in the document]
Comment (rgao) [reply src/styles/contact/_contactimage.scss]: Ah right, I forgot to ask about where to get the images. I'll get that fixed

File: src/components/contact/ContactForm.jsx [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+      message: '',
+      errorMessage: '',
+    };
+  }
+
+  onInputChange = event => {
+    this.setState({ [event.target.name]: event.target.value });
+  };
+
+  handleSubmit = event => {
```
[the remark this answers is not in the document]
Comment (rgao) [reply src/components/contact/ContactForm.jsx]: I'll work on it

File: package.json [hunk 3]
```diff
@@ -17,6 +17,7 @@
     "dotenv": "^8.2.0",
     "dotenv-webpack": "^1.7.0",
     "eslint-import-resolver-webpack": "^0.12.1",
+    "express": "^4.17.1",
```
[the remark this answers is not in the document]
Comment (rgao) [reply package.json]: I'll remove it. I think I added it by accident

**Comment by rgao** [comment]
Alright, did all of the above

**Comment by adamkendis** [comment]
Looks good @rgao . We can beef up the validation or use some validation library before launch if necessary. Okay to update this branch with latest dev so I can merge?

**Comment by rgao** [comment]
Alright, updated

**Comment by adamkendis** [comment]
Sorry, going to hold off on merging for now. I pushed the github code in PR #481 . Could you finish up the form to send the data to /feedback? The request body should look like:

**Comment by rgao** [comment]
@adamkendis @sellnat77 
OK I pushed the redux code for contacts. It's my first time dealing with redux so it probably has errors or inconsistencies in it. Could you please take a look at it?

**Comment by jmensch1** [comment]
hey @rgao, one other thing. Sorry to ask for more work, but it's pretty easy.

Right now all of the styles in `.contactform.scss` file will apply globally, because they aren't nested within anything. So you'll end up affecting the styles of every input, label and textarea in the entire frontend. 

All you need to do to avoid that is nest all of the styles in that file under the `.form-container` class, and then the styles will only apply to elements within that div. And maybe rename `.form-container` to something more specific, because that's a pretty generic name and if anyone else tried to use it they'd be getting all of your styles.

**Comment by rgao** [comment]
Alright, working on all these changes

**Comment by sellnat77** [comment]
i defer to @jmensch1 and @adamkendis haha i looked it over though, once they approve we can merge this in

**Comment by brodly** [comment]
The comparison tool made a lot of changes so once you resolve your merge conflicts we can push this to dev and catch any issues during QA.

File: src/components/contact/ContactForm.jsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+      this.setState({
+        disableButton: false,
+        buttonText: 'Submite',
+      });
+    } else {
+      this.setState({
+        disableButton: true,
+      });
+    }
+  }
```
[the remark this answers is not in the document]
Comment (rgao) [reply src/components/contact/ContactForm.jsx:58]: forgot to delete this deprecated code

File: src/redux/rootSaga.js [hunk 5]
```diff
@@ -141,7 +154,19 @@ function* getPinData(action) {
   }
 }
 
+function* getContactData(action) {
+  try {
+    const email = action.payload;
```
[the remark this answers is not in the document]
Comment (rgao) [reply src/redux/rootSaga.js:159]: added the request (form fields) as payload

File: src/redux/reducers/data.js [hunk 6]
```diff
@@ -107,6 +124,33 @@ export default (state = initialState, action) => {
...
     }
+    case types.GET_CONTACT_REQUEST:
+      return {
+        state,
+      };
+    case types.GET_CONTACT_SUCCESS:
+      return {
+        ...state,
+        error: null,
+        isLoading: false,
```
[the remark this answers is not in the document]
Comment (rgao) [reply src/redux/reducers/data.js:135]: done

**Comment by rgao** [comment]
there are two linting errors but that's due to my linting dependencies being broken

**Comment by rgao** [comment]
Alright that's my final push for tonight. @adamkendis The linting issues should be solved; I think it's better if you take over any potential problems with redux here

**Comment by adamkendis** [comment]
@brodly @jmensch1 @sellnat77 @rgao 
The Contact page is finished. A few things:
- Fixed contact form stylesheet so we shouldn't have unintentional global styles.
- Added loader, error display, and success message display to contact form.
- Added email validation.
- Back end: fixed feedbackService error handling. I wasn't properly returning the errors to the client, so the browser would get a 200 even on error.

**Pull request merged**